bindings/python/scripts/clang_utils.py

```diff
@@ -0,0 +1,138 @@
+import inspect
+import clang.cindex as clang
+
+
+def getmembers_static(object, predicate=None):
+    """
+    Return all members of an object as (name, value) pairs sorted by name.
+    Optionally, only return members that satisfy a given predicate.
+    """
+
+    """
+    - `getmembers` function (from the inspect module) triggers execution instead of doing static analysis.
+    - This leads to errors, particularly on properties of classes in cindex.py, which causes segmentation errors or raises an Exception if a particular condition is not satisfied.
+    - To curb this, we need a fetch the members statically.
+    - The inspect module doesn't provide a function to do so. So we define a custom function based on the one in the inspect module.
+    
+    diff: We replace the `getattr` function call with a `getattr_static` function call in the function definition.
+    """
+
+    if inspect.isclass(object):
+        mro = (object,) + inspect.getmro(object)
+    else:
+        mro = ()
+    results = []
+    processed = set()
+    names = dir(object)
+    # :dd any DynamicClassAttributes to the list of names if object is a class;
+    # this may result in duplicate entries if, for example, a virtual
+    # attribute with the same name as a DynamicClassAttribute exists
+    try:
+        for base in object.__bases__:
+            for k, v in base.__dict__.items():
```

Comment:
```py
base_members = filter(lambda k,v: isinstance(v, types.DynamicClassAttribute),
                      base.__dict__.items())
names.extend(base_memebrs)
```

Some similar format, basically show case the filter and append aspect separately

Reply:
Done, thanks for the suggestion.